File: backend/app/schemas/booking.py
"""Import necessary modules"""
from datetime import datetime
from pydantic import BaseModel, Field

class BookingCreate(BaseModel):
    """Validation for creating a booking"""
    name: str = Field(..., min_length=2)
    phone: str = Field(..., min_length=11, max_length=15)
    bus_provider: str
    from_district: str
    to_district: str
    dropping_point: str
    price: float
    travel_date: str
    travel_time: str

class BookingResponse(BaseModel):
    """Validation for booking response"""
    id: int
    name: str
    phone: str
    bus_provider: str
    from_district: str
    to_district: str
    dropping_point: str
    price: float
    traveL_date: str
    travel_time: str
    booking_date: datetime
    status: str

File: frontend/src/services/types/booking.ts
export interface BookingCreateRequest {
  name: string;
  phone: string;
  bus_provider: string;
  from_district: string;
  to_district: string;
  dropping_point: string;
  price: number;
  travel_date: string;
  travel_time: string;
}

export interface Booking {
  id: number;
  name: string;
  phone: string;
  bus_provider: string;
  from_district: string;
  to_district: string;
  dropping_point: string;
  price: number;
  travel_date: string;
  travel_time: string;
  booking_date: string;
  status: 'confirmed' | 'canceled';
}

export interface CancelBookingRequest {
  phone: string;
  travel_date: string;
  travel_time: string;
  bus_provider: string;
  from_district: string;
  to_district: string;
  dropping_point: string;
}